src/Link.js

```diff
@@ -1,7 +1,108 @@
-import BaseLink from './BaseLink';
-import withRouter from './withRouter';
+import useEventCallback from '@restart/hooks/useEventCallback';
+import React from 'react';
+import warning from 'tiny-warning';
 
-const Link = withRouter(BaseLink);
-Link.displayName = 'Link';
+import useRouter from './useRouter';
+
+function Link({
+  as: Component = 'a',
+  to,
+  activeClassName,
+  activeStyle,
+  activePropName,
+  match: propsMatch,
+  router: propsRouter,
+  exact = false,
+  onClick,
+  target,
+  ...props
+}) {
+  const { router, match } = useRouter() || {
+    match: propsMatch,
+    router: propsRouter,
+  };
+
+  const handleClick = useEventCallback((event) => {
+    if (onClick) {
+      onClick(event);
+    }
+
+    // Don't do anything if the user's onClick handler prevented default.
+    // Otherwise, let the browser handle the link with the computed href if the
+    // event wasn't an unmodified left click, or if the link has a target other
+    // than _self.
+    if (
+      event.defaultPrevented ||
+      event.metaKey ||
+      event.altKey ||
+      event.ctrlKey ||
+      event.shiftKey ||
+      event.button !== 0 ||
+      (target && target !== '_self')
+    ) {
+      return;
+    }
+
+    event.preventDefault();
+
+    // FIXME: When clicking on a link to the same location in the browser, the
+    // actual becomes a replace rather than a push. We may want the same
```

Comment:
never heard of the term "actual" like this

though it could be correct for all I know